src/test/java/school/hei/haapi/integration/CourseIT.java

```diff
@@ -0,0 +1,352 @@
+import school.hei.haapi.SentryConf;
+import school.hei.haapi.endpoint.rest.api.TeachingApi;
+import school.hei.haapi.endpoint.rest.api.UsersApi;
+import school.hei.haapi.endpoint.rest.client.ApiClient;
+import school.hei.haapi.endpoint.rest.client.ApiException;
+import school.hei.haapi.endpoint.rest.model.Course;
+import school.hei.haapi.endpoint.rest.model.CourseDirection;
+import school.hei.haapi.endpoint.rest.model.CrupdateCourse;
+import school.hei.haapi.endpoint.rest.model.UpdateStudentCourse;
+import school.hei.haapi.endpoint.rest.security.cognito.CognitoComponent;
+import school.hei.haapi.integration.conf.AbstractContextInitializer;
+import school.hei.haapi.integration.conf.TestUtils;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+import static school.hei.haapi.endpoint.rest.model.CourseStatus.LINKED;
+import static school.hei.haapi.endpoint.rest.model.CourseStatus.UNLINKED;
+import static school.hei.haapi.integration.conf.TestUtils.COURSE1_ID;
+import static school.hei.haapi.integration.conf.TestUtils.COURSE2_ID;
+import static school.hei.haapi.integration.conf.TestUtils.COURSE3_ID;
+import static school.hei.haapi.integration.conf.TestUtils.COURSE4_ID;
+import static school.hei.haapi.integration.conf.TestUtils.COURSE5_ID;
+import static school.hei.haapi.integration.conf.TestUtils.MANAGER1_TOKEN;
+import static school.hei.haapi.integration.conf.TestUtils.STUDENT1_ID;
+import static school.hei.haapi.integration.conf.TestUtils.STUDENT1_TOKEN;
+import static school.hei.haapi.integration.conf.TestUtils.STUDENT2_ID;
+import static school.hei.haapi.integration.conf.TestUtils.TEACHER1_TOKEN;
+import static school.hei.haapi.integration.conf.TestUtils.TEACHER4_ID;
+import static school.hei.haapi.integration.conf.TestUtils.anAvailableRandomPort;
+import static school.hei.haapi.integration.conf.TestUtils.assertThrowsApiException;
+import static school.hei.haapi.integration.conf.TestUtils.assertThrowsForbiddenException;
+import static school.hei.haapi.integration.conf.TestUtils.setUpCognito;
+import static school.hei.haapi.integration.conf.TestUtils.teacher1;
+import static school.hei.haapi.integration.conf.TestUtils.teacher2;
+import static school.hei.haapi.integration.conf.TestUtils.teacher3;
+import static school.hei.haapi.integration.conf.TestUtils.teacher4;
+
+@SpringBootTest(webEnvironment = RANDOM_PORT)
+@Testcontainers
+@ContextConfiguration(initializers = CourseIT.ContextInitializer.class)
+@AutoConfigureMockMvc
+class CourseIT {
+  @MockBean
+  private SentryConf sentryConf;
+  @MockBean
+  private CognitoComponent cognitoComponentMock;
+
+  private static ApiClient anApiClient(String token) {
+    return TestUtils.anApiClient(token, CourseIT.ContextInitializer.SERVER_PORT);
+  }
+
+  static Course course1() {
+    Course course = new Course();
```

Comment:
Return and inline variable instead and do it for all resources.

For example : 

```
new Course()
.id(...)
.code(...)
```